static/app/views/settings/account/accountIdentities.tsx

```diff
@@ -34,54 +43,146 @@ class AccountIdentities extends AsyncView<Props, State> {
     return t('Identities');
   }
 
-  handleDisconnect = (identity: Identity) => {
-    const {identities} = this.state;
-
-    this.setState(
-      state => {
-        const newIdentities = state.identities?.filter(({id}) => id !== identity.id);
-
-        return {
-          identities: newIdentities ?? [],
-        };
-      },
-      () =>
-        disconnectIdentity(identity).catch(() => {
-          this.setState({
-            identities,
-          });
-        })
+  renderItem = (identity: UserIdentityConfig) => {
+    return (
+      <IdentityPanelItem key={`${identity.category}:${identity.id}`}>
+        <InternalContainer>
+          <PluginIcon pluginId={identity.provider.key} size={36} />
+          <IdentityText isSingleLine={!identity.dateAdded}>
+            <IdentityName>{identity.provider.name}</IdentityName>
+            {identity.dateAdded && <IdentityDateTime date={moment(identity.dateAdded)} />}
+          </IdentityText>
+        </InternalContainer>
+        <InternalContainer>
+          <TagWrapper>
+            <Tag
+              isOrgName={false}
+              label={
+                identity.category === UserIdentityCategory.SOCIAL_IDENTITY
+                  ? t('App Integration')
+                  : identity.category === UserIdentityCategory.GLOBAL_IDENTITY
+                  ? t('Sign-In Identity')
+                  : identity.category === UserIdentityCategory.ORG_IDENTITY
+                  ? t('Organization Identity')
+                  : ''
+              }
+            />
+            {identity.organization && (
+              <Tag isOrgName label={identity.organization.slug} />
+            )}
+          </TagWrapper>
+
+          {this.renderButton(identity)}
+        </InternalContainer>
+      </IdentityPanelItem>
     );
   };
 
+  renderButton(identity: UserIdentityConfig) {
+    return identity.status === UserIdentityStatus.CAN_DISCONNECT ? (
+      <Confirm
+        onConfirm={() => this.handleDisconnect(identity)}
+        priority="danger"
+        confirmText={t('Disconnect')}
+        message={
+          <React.Fragment>
+            <Alert type="error" icon={<IconFlag size="md" />}>
+              {t(`Disconnect Your ${identity.provider.name} Identity?`)}
```

Comment:
Unfortunately you can't do translations like this. interpolation in a translation string is a nono.

What you want is `tct('Disconnect Your [provider] Identity', {provider: identity.provider.name})`

Reply:
Ooh, thanks. I thought maybe it would be okay because there is a closed set of provider names, but I meant to ask. Will fix.

Reply:
Unfortunately the static analysis that extracts locale strings for translation is not that smart 😬